wgpu-hal/src/gles/command.rs

```diff
@@ -30,6 +30,46 @@ pub(super) struct State {
     dirty_vbuf_mask: usize,
 }
 
+impl crate::CopyExtent {
```

Comment:
All these non-local impls seem misleading -- it exists at the top level, but only if the gles feature is enabled. I think these should just be implemented at the top level and the other backends can just not use them.

Reply:
Actually, I really liked how they allow us to structure the code. The `impl` itself doesn't have visibility, but the function inside the `impl` does. So if this function is `pub(super)`, it means it's only visible in `gles` sub-module, and not the top-level like you assumed.

Reply:
Now I see what you mean, the code isn't backend-specific and should be moved higher up. I agree. Was implementing GLES last and didn't want to re-touch all the previous code.

Reply:
That is an interesting side effect, I had no idea you could do that at all.